1.raspberry-pi-stuff/octoberry/octoberry/__main__.py

```diff
@@ -1,98 +1,125 @@
+
+    Note: if you would like to read more about configuring this device you can follow this:
     https://download.wezhan.cn/contents/sitefiles2041/10205132/files/473369..pdf?response-content-disposition=inline%3Bfilename%2A%3Dutf-8%27%27FissaiD-EH301-V8-OnLine.pdf&response-content-type=application%2Fpdf&auth_key=1695929334-f2ca675f6c4842eda7036f465ffdbeac-0-cca5608b286471bd42819337066a0dbb
 
     Returns:
-        str: RFID
-    '''
-    data_read = input()
-    # will look something like this: ABC12345678XYZ
-    # where 123 is FC and 45678 is ID
-    match = re.match(r'^ABC(?P<FC>\d{3})(?P<ID>\d{5})XYZ$', data_read)
-
-    if not match:
-        print("Invalid RFID scan")
+        tuple of (fc, card) if RFID is valid, None otherwise
+    """
+    data_read = list(keyboard.get_typed_strings(keyboard.record(until="enter")))[0]
+    if len(data_read) != 10:
+        logger.error(f"RFID is not 10 digits, RFID: {data_read}")
         return None
     
-    fc, id = match.group('FC'), match.group('ID')
-    print(f"FC: {fc}, ID: {id}")
-    return fc, id
+    # will be 10 digits number in the base of 10
+    read_value_value = int(data_read, 10)
+
+    # the format being used here is 8h-10d in the documentation
+    fc = (read_value_value >> 16) & 0xFF
+    card = read_value_value & 0xFFFF
+
+    return fc, card
+
 
 def main():
+    formatter = logging.Formatter(
+        fmt="%(asctime)s %(name)s.%(levelname)s: %(message)s",
+        datefmt="%Y.%m.%d %H:%M:%S",
+    )
+
+    logger.info(f"Working directory: {os.getcwd()}")
+
     while True:
-        print("Waiting for RFID scans...")
+        logger.info("Waiting for RFID scans...")
         rfid_return_value = read_rfid()
         if not rfid_return_value:
             continue
-        fc, id = rfid_return_value
-        
-        print("RFID scanned, fd: {fc}, id: {id}")
-
-        response = requests.get(f"{BACKEND_URL}/rfid_check", params={'fc': fc, 'id': id})
-        match response.status_code:
-            case requests.codes.ok:
-                print("RFID is valid")
-                sound_url = response.json()['sound_url']
-                play_sound(sound_url)
-            case requests.codes.unauthorized:
-                print("RFID is not registered") 
-                # TODO: play sound that says "RFID is not registered... your registration color is X"
-            case _:
-                raise Exception(f"Failed to check RFID, status code: {response.status_code}")
-
-
-
-if __name__ == '__main__':
-    main()
\ No newline at end of file
+        fc, card = rfid_return_value
+
+        logger.info(f"RFID scanned, fd: {fc}, card: {card}")
+
+        response = requests.post(
+            f"{BACKEND_URL}/api/scan",
+            json={"fc": fc, "card": card},
+            headers={"x-api-key": os.environ["API_TOKEN"]},
+        )
+        if response.status_code == requests.codes.ok:
+            logger.info("RFID is valid")
+            sound_urls = response.json()["sound_urls"]
```

Comment:
@gregsadetsky just thought that from the rpi perspective there is no indication whether the nfc tag is registered or not.

we're ok with that?